### PR TITLE
[ALLUXIO-3243] Optimize concatPath

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-3243

This creates a StringBuilder of the correct size, so it does not have to be constantly resized.

### CODE REVIEW

**Comment by gpang** [comment]
Jenkins, test this please

File: core/common/src/main/java/alluxio/util/io/PathUtils.java [hunk 1]
```diff
@@ -50,6 +48,36 @@ public static String cleanPath(String path) throws InvalidPathException {
...
+   * <p>
+   * Note that empty element in base or paths is ignored.
+   *
+   * @param base base path
+   * @param path path element to concatenate
+   * @return joined path
+   * @throws IllegalArgumentException if base or paths is null
+   */
+  public static String concatPath(Object base, Object path) throws IllegalArgumentException {
+    Preconditions.checkArgument(base != null, "Failed to concatPath: base is null");
```
Comment (apc999) [inline core/common/src/main/java/alluxio/util/io/PathUtils.java]: nit `checkNotNull`

**Comment by apc999** [comment]
shall we also provide the version of `concatPath` with 3, 4,... args?

**Comment by gpang** [comment]
Jenkins, test this please

File: core/common/src/main/java/alluxio/util/io/PathUtils.java (hunk 1 above)
Comment (gpang) [reply core/common/src/main/java/alluxio/util/io/PathUtils.java]: Done

**Pull request merged**